lib/Conversion/TorchToLinalg/Uncategorized.cpp

```diff
@@ -2565,10 +2565,29 @@ class ConvertAtenGridSamplerOp : public OpConversionPattern<AtenGridSamplerOp> {
       return res;
     };
 
+    auto lambdaBorder = [&](OpBuilder &b, Location loc, Value x,
+                            Value SizeSubOne) -> Value {
+      Value xMaxZero = b.create<arith::MaximumFOp>(loc, x, zeroFloat);
+      return b.create<arith::MinimumFOp>(loc, xMaxZero, SizeSubOne);
+    };
+
+    auto lambdaPadding = [&](OpBuilder &b, Location loc, Value paddingMode,
+                             Value x, Value SizeSubOne) -> Value {
+      Value border = lambdaBorder(b, loc, x, SizeSubOne);
+      Value zeroInt =
+          b.create<arith::ConstantOp>(loc, b.getIntegerAttr(int64type, 0));
+      Value isZero = b.create<arith::CmpIOp>(loc, arith::CmpIPredicate::eq,
+                                             paddingMode, zeroInt);
```

Comment:
The padding mode is always a constant (coming from ONNX, at least). We should be able to get the `paddingModeInt` by `matchPattern(paddingMode, m_TorchConstantInt(&paddingModeInt))`. This way, we can avoid writing IR to check conditions that can be determined statically at compile time. Since the select op and cmp op will likely get folded anyway, this is not hugely problematic, but on principal I'd prefer to generate the simplest possible IR from the get-go.

Reply:
Yes, I'm evaluating it at compile time now.